server/src/main/java/com/lunchtime/service/RestaurantService.java

```diff
@@ -0,0 +1,61 @@
+package com.lunchtime.service;
+
+import com.lunchtime.models.Restaurant;
+import com.lunchtime.repository.RestaurantRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+import java.util.Optional;
+
+@Service
+public class RestaurantService {
+
+    @Autowired
+    private RestaurantRepository restaurantRepository;
+
+    public Restaurant save(Restaurant restaurant) {
+        return restaurantRepository.save(restaurant);
+    }
+
+    public Page<Restaurant> findAll(Pageable pageable) {
+        return restaurantRepository.findAll(pageable);
+    }
+
+    public Optional<Restaurant> findById(Long id) {
+        return restaurantRepository.findById(id);
+    }
+
+    public Restaurant update(Restaurant newRestaurant) {
+        return findById(newRestaurant.getId())
+            .map(restaurant -> {
+                restaurant.setName(newRestaurant.getName());
+                restaurant.setEmail(newRestaurant.getEmail());
+                restaurant.setTextAddress(newRestaurant.getTextAddress());
+                restaurant.setWebsite(newRestaurant.getWebsite());
+                restaurant.setDescription(newRestaurant.getDescription());
+                restaurant.setWorkingTime(newRestaurant.getWorkingTime());
+                restaurant.setMenuId(newRestaurant.getMenuId());
+                restaurant.setOwnerId(newRestaurant.getOwnerId());
+                restaurant.setTables(newRestaurant.getTables());
+                restaurant.setLongitude(newRestaurant.getLongitude());
+                restaurant.setLatitude(newRestaurant.getLatitude());
```

Comment:
It doesn't look very good. Can someone know a better solution?

Reply:
For sure, Hibernate has prepared something useful for us. [**Check this**](https://stackoverflow.com/questions/2398981/effective-entity-update-in-hibernate), please

Reply:
Thanks for answer. But `dynamicUpdate=true` doesn't help in this case. 
This code looks bad because developer must look after 2 places in codebase. When entity Restaurant is updated then method update(restaurant) must be updated too. 
I can't save newRestaraunt object directly, because it comes from api request and can contain fields like createdAt with wrong data (or other wrong fields).

Reply:
I think we should hold this place for future and return here. There must be more flexible way to update our entity. Now we can't imagine Hibernate's power, but we will consider to improve it later. It's ok.